pkg/cmd/util/clientcmd/factory.go

```diff
@@ -234,17 +238,16 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	}
 	kLogsForObjectFunc := w.Factory.LogsForObject
 	w.LogsForObject = func(object, options runtime.Object) (*kclient.Request, error) {
-		oc, _, err := w.Clients()
-		if err != nil {
-			return nil, err
-		}
-
 		switch t := object.(type) {
 		case *deployapi.DeploymentConfig:
 			dopts, ok := options.(*deployapi.DeploymentLogOptions)
 			if !ok {
 				return nil, errors.New("provided options object is not a DeploymentLogOptions")
 			}
+			oc, err := w.clients.ClientForVersion("")
```

Comment:
why not continue to call w.Clients()? ClientForVersion("") seems like something we might have to revisit...